src/core/alloc_func.hpp

```diff
@@ -125,7 +125,7 @@ static inline T *ReallocT(T *t_ptr, size_t num_elements)
 	/* Ensure the size does not overflow. */
 	CheckAllocationConstraints<T>(num_elements);
 
-	t_ptr = (T*)realloc(t_ptr, num_elements * sizeof(T));
+	t_ptr = (T*)realloc(static_cast<void *>(t_ptr), num_elements * sizeof(T));
```

Comment:
This is dangerous as it silences warnings that might actually be appropriate. E.g. it might prevent a compiler from warning against realloc'ing an array allocated with `operator new[]`. If ´ReallocT<T>` causes warnings at any call sites those call sites should be fixed to use the appropriate way of changing size.

Reply:
Yup, I realised this as I implemented it. I did check all the existing warnings before I changed it, so I don't believe there's any issues with it currently. I'm open to suggestions for how to solve it though - some sort of template specialisation incantation, maybe?

Of course, ideally all need for realloc would be replaced with std::vector or similar anyway :p